Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.cpp

```diff
@@ -251,6 +251,49 @@ int64_t WebPlatformStrategies::setStringForType(const String& string, const Stri
     return newChangeCount;
 }
 
+static void collectFrameWebArchives(WebCore::FrameIdentifier frameIdentifier, HashMap<FrameIdentifier, Ref<WebCore::LegacyWebArchive>>& archives, Vector<WebCore::FrameIdentifier>& remoteFrameIdentifiers)
+{
+    RefPtr webFrame = WebFrame::webFrame(frameIdentifier);
+    if (!webFrame)
+        return;
+
+    for (RefPtr<Frame> frame = webFrame->coreFrame(); frame; frame = frame->tree().traverseNext()) {
```

Comment:
As we discussed on slack, I think we wanna specify `stayWithin` here. Otherwise, we iterate over the subsequent frames multiple times.